src/contents/ui/components/SongAndArtistText.qml

```diff
@@ -6,45 +6,119 @@ import org.kde.kirigami as Kirigami
 ColumnLayout {
     id: root
 
+    enum TextPosition {
+        Hidden,
+        FirstLine,
+        SecondLine,
+        ThirdLine
+    }
+
     property var maxWidth: undefined
     property var scrollingBehaviour: undefined
     property var scrollingSpeed: undefined
     property var scrollingResetOnPause: undefined
     property var scrollingEnabled: undefined
     property var forcePauseScrolling: undefined
-    property bool splitSongAndArtists: false
+
+    // 0 is hidden, 1 is first line, 2 is second line
+
+    property int titlePosition: SongAndArtistText.TextPosition.FirstLine
+    property int artistsPosition: SongAndArtistText.TextPosition.FirstLine
+    property int albumPosition: SongAndArtistText.TextPosition.Hidden
+
     property font textFont: Kirigami.Theme.defaultFont
     property font boldTextFont: Qt.font(Object.assign({}, textFont, {weight: Font.Bold}))
     property string color: Kirigami.Theme.textColor
     property string title
     property string artists
+    property string album
     property int textAlignment: Qt.AlignHCenter
 
     spacing: 0
 
+
+    property var firstLineArray: {
+        const arr = [];
+
+        if (artistsPosition == SongAndArtistText.TextPosition.FirstLine) arr.push(root.artists);
+        if (titlePosition   == SongAndArtistText.TextPosition.FirstLine) arr.push(root.title);
+        if (albumPosition   == SongAndArtistText.TextPosition.FirstLine) arr.push(root.album);
+
+        return arr;
+    }
+
+    property var secondLineArray: {
+        const arr = [];
+
+        if (artistsPosition == SongAndArtistText.TextPosition.SecondLine) arr.push(root.artists);
+        if (titlePosition   == SongAndArtistText.TextPosition.SecondLine) arr.push(root.title);
+        if (albumPosition   == SongAndArtistText.TextPosition.SecondLine) arr.push(root.album);
+
+        return arr;        
+    }
+
+    property var thirdLineArray: {
+        const arr = [];
+
+        if (artistsPosition == SongAndArtistText.TextPosition.ThirdLine) arr.push(root.artists);
+        if (albumPosition   == SongAndArtistText.TextPosition.ThirdLine) arr.push(root.album);
+        if (titlePosition   == SongAndArtistText.TextPosition.ThirdLine) arr.push(root.title);
+
+        return arr;   
+    }
+
+    property string finalFirstText:  firstLineArray.filter((x) => x).join(" - ")
+    property string finalSecondText: secondLineArray.filter((x) => x).join(" - ")
+    property string finalThirdText: thirdLineArray.filter((x) => x).join(" - ")
+
+    // first row of text (the only row, if there is only one)
     ScrollingText {
-        visible: root.splitSongAndArtists
+        // visible only when necessary
+        visible: text.length !== 0
         overflowBehaviour: root.scrollingBehaviour
         font: root.boldTextFont
```

Comment:
To maintain the previous behaviour, we should make the first line bold only when there is the second line:
```suggestion
        font: finalSecondText.length > 0 ? root.boldTextFont : root.textFont;
```